modules/report/includes/commerce_pos_report.pages.inc

```diff
@@ -1005,3 +1004,300 @@ function commerce_pos_report_build_journal_role_table(array $filters = array())
+      $table_row              = array();
+      $transactions           = $cashier_transactions['cashier']['transactions'];
+      $currency_code          = $cashier_transactions['cashier']['currency_code'];
+      $net_sales              = $cashier_transactions['transaction']['net_sales'];
+      $net_sold_items         = ($cashier_transactions['transaction']['purchase_count'] - $cashier_transactions['transaction']['return_count']);
+      $unit_per_transaction   = round(($net_sold_items / $transactions), 2);
+      $dollar_per_transaction = ($net_sales / $transactions);
+      $table_row[]            = $cashier_transactions['cashier']['id'];
+      $table_row[]            = $cashier_transactions['cashier']['name'];
+      $table_row[]            = commerce_currency_format($net_sales, $currency_code);
+      $table_row[]            = $transactions;
+      $table_row[]            = $net_sold_items;
+      // Unit per Trans (UPT)
+      $table_row[] = $unit_per_transaction;
+      $table_row[] = commerce_currency_format($dollar_per_transaction, $currency_code);
+      // Table feet with total.
+      $totals['net_sales'] += $net_sales;
+      $totals['transactions'] += $transactions;
+      $totals['item_sold'] += $net_sold_items;
+      $totals['unit_per_transaction'] += $unit_per_transaction;
+      $totals['dollar_per_transaction'] += $dollar_per_transaction;
+      $rows[] = array(
+        'data' => $table_row,
+        'class' => '',
+      );
+    }
+
+    $rows[] = array(
+      '',
+      '',
+      commerce_currency_format($totals['net_sales'], $currency_code),
+      $totals['transactions'],
+      $totals['item_sold'],
+      $totals['unit_per_transaction'],
+      commerce_currency_format($totals['dollar_per_transaction'], $currency_code),
+    );
+
+    $table = array(
+      '#theme' => 'table',
+      '#rows' => $rows,
+      '#header' => $header,
+      '#attached' => array(
+        'js' => array(
+          drupal_get_path('module', 'commerce_pos_report') . '/js/commerce_pos_report.journal_role.js',
+        ),
+        'library' => array(
+                array('system', 'drupal.ajax'),
+        ),
+      ),
+    );
+    return $table;
+  }
+  else {
+    return array(
+      '#markup' => t('No transactions found'),
+    );
+  }
+}
+
+/**
+ * Callback for the KPI form/report.
+ */
+function commerce_pos_report_kpi_report($form, &$form_state) {
+  global $user;
+  $form['#tree'] = TRUE;
+  $form['header'] = array(
+    '#markup' => theme('commerce_pos_header', array('account' => $user)),
+  );
+
+  $query_filter_defaults = array(
+    'results_per_page' => COMMERCE_POS_REPORT_KPI_REPORT_DEFAULT_ITEMS_PER_PAGE,
+  );
+
+  $query_params = drupal_get_query_parameters() + $query_filter_defaults;
+
+  $form['filters'] = array(
+    '#type' => 'container',
+  );
+
+  if (!isset($form_state['results_container_id'])) {
+    $form_state['results_container_id'] = 'commerce-pos-report-journal-role-container';
```

Comment:
Again I wonder if we should refactor this in both reports and the JS file to be more report-agnostic...